src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1524,10 +1524,8 @@ public sealed override bool IsTupleCompatible(out int tupleCardinality)
             return false;
         }
 
-        // PROTOTYPE: Should be abstract.
-        internal virtual NamedTypeSymbol AsNativeInt(bool asNativeInt) => this;
+        internal virtual NamedTypeSymbol AsNativeInt(bool asNativeInt) => throw ExceptionUtilities.Unreachable;
 
-        // PROTOTYPE: Should be abstract.
         internal virtual bool IsNativeInt => false;
```

Comment:
>internal virtual bool IsNativeInt => false; [](start = 8, length = 43)

I think we would want this to be abstract to make sure it is properly implemented by all leaf subclasses.

Reply:
I've added back the "should be abstract" PROTOTYPE comment.

---
In reply to: [370288534](https://github.com/dotnet/roslyn/pull/40994#discussion_r370288534) [](ancestors = 370288534)